components/dashboard/src/components/Modal.tsx

```diff
@@ -1,20 +1,47 @@
+import { Disposable, DisposableCollection } from "@gitpod/gitpod-protocol";
+
 export default function Modal(props: {
     children: React.ReactChild[] | React.ReactChild,
     visible: boolean,
     closeable?: boolean,
     className?: string,
-    onClose: () => void
+    onClose: () => void,
+    onEnter?: () => boolean
 }) {
+    const disposable = new DisposableCollection();
+    const close = () => {
+        disposable.dispose();
+        props.onClose();
+    }
     if (!props.visible) {
         return null;
     }
-    setTimeout(() => window.addEventListener('click', props.onClose, { once: true }), 0);
+    setTimeout(() => {
+        const keyHandler = (k: globalThis.KeyboardEvent) => {
+            if (!k.bubbles) {
+                if (k.key === 'Escape') {
+                    close();
+                }
+                if (k.key === 'Enter') {
+                    if (props.onEnter) {
+                        if (props.onEnter() === false) {
+                            return;
+                        }
+                    }
+                    close();
+                    k.stopPropagation();
+                }
+            }
+        }
+        window.addEventListener('keydown', keyHandler, { capture: true });
+        disposable.push(Disposable.create(()=> window.removeEventListener('keydown', keyHandler)));
+    }, 0);
     return (
-        <div className="fixed top-0 left-0 bg-black bg-opacity-70 z-50 w-screen h-screen" >
+        <div className="fixed top-0 left-0 bg-black bg-opacity-70 z-50 w-screen h-screen" onClick={close}>
```

Comment:
I believe, `onClick={close}` here will trigger `close` even if you click-select anything within the modal.

Reply:
It does not, because I stop.propagation below

Reply:
I see